readthedocs/search/utils.py

```diff
@@ -168,3 +169,38 @@ def _get_sorted_results(results, source_key='_source'):
     ]
 
     return sorted_results
+
+
+def record_search_query(project_slug, version_slug, query, total_results):
+    """Record search query in database."""
+    if not project_slug or not version_slug or not query or not total_results:
```

Comment:
Should this function be a celery task?
Every search has to create/update object in database before showing results to the user.
I am not sure on how much this will impact the speed.
But I think that it will be better to have this as a celery task -- so that it can create the objects asynchronously and we have the results at the same speed.

Reply:
Yea, having it be out of the direct query response will be best.